nixos/modules/services/web-apps/opencloud.nix

```diff
@@ -0,0 +1,212 @@
+        type = types.nullOr types.path;
+        default = null;
+        example = "/var/lib/opencloud/config";
+        description = ''
+          Path to directory containing OpenCloud config file.
+
+          Example config can be generated by `opencloud init --config-path fileName --admin-password "adminPass"`.
+          Add `--insecure true` if SSL certificates are generated and managed externally (e.g. using OpenCloud behind a reverse proxy).
+
+          Note: This directory must contain at least a `opencloud.yaml`. Ensure
+          [user](#opt-services.opencloud.user) has read/write access to it. In some
+          circumstances you may need to add additional OpenCloud configuration files (e.g.,
+          `proxy.yaml`) to this directory.
+        '';
+      };
+
+      environmentFile = lib.mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/run/keys/opencloud.env";
+        description = ''
+          An environment file as defined in {manpage}`systemd.exec(5)`.
+
+          Configuration provided in this file will override those from [configDir](#opt-services.opencloud.configDir)/opencloud.yaml.
+        '';
+      };
+
+      user = lib.mkOption {
+        type = types.str;
+        default = defaultUser;
+        example = "yourUser";
+        description = ''
+          The user to run OpenCloud as.
+          By default, a user named `${defaultUser}` will be created whose home
+          directory is [stateDir](#opt-services.opencloud.stateDir).
+        '';
+      };
+
+      group = lib.mkOption {
+        type = types.str;
+        default = defaultGroup;
+        example = "yourGroup";
+        description = ''
+          The group to run OpenCloud under.
+          By default, a group named `${defaultGroup}` will be created.
+        '';
+      };
+
+      address = lib.mkOption {
+        type = types.str;
+        default = "127.0.0.1";
+        description = "Web interface address.";
+      };
+
+      port = lib.mkOption {
+        type = types.port;
+        default = 9200;
+        description = "Web interface port.";
+      };
+
+      url = lib.mkOption {
+        type = types.str;
+        default = "https://localhost:9200";
+        example = "https://some-hostname-or-ip:9200";
+        description = "Web interface address.";
+      };
+
+      stateDir = lib.mkOption {
+        default = "/var/lib/opencloud";
+        type = types.str;
+        description = "OpenCloud data directory.";
+      };
+
+      environment = lib.mkOption {
+        type = types.attrsOf types.str;
+        default = { };
+        description = ''
+          Extra config options.
+
+          Note that all the attributes here will be copied to /nix/store/ and will be world readable. Options like *_PASSWORD or *_SECRET should be part of     [environmentFile](#opt-services.opencloud.environmentFile) instead, and are only provided here for illustrative purpose.
```

Comment:
This attributes are not copied, that is technically wrong. They are written to a file which is placed in the nix store.

Don't provide examples for illustration purposes if they are bad advice.